### PR TITLE
Babel transpiler

### PR DESCRIPTION
**Pull request by dabrau**
use babel because es6 is pretty dope

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,14 +15,21 @@
...
     "babelify": "^7.2.0",
-    "envify": "^4.0.0"
+    "browserify": "^10.2.4",
+    "browserify-incremental": "^3.0.1",
+    "envify": "^4.0.0",
+    "eslint": "^3.15.0",
+    "eslint-plugin-react": "^6.9.0",
+    "redux-logger": "^2.8.1"
+  },
+  "scripts": {
```
Comment (dabrau) [inline package.json]: "npm run lint" to lint all the js files

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+app/assets/javascripts/application.js.coffee
+app/assets/javascripts/model.js.coffee
+app/assets/javascripts/jquery-mask.js
+app/assets/javascripts/datetimepicker/jquery.datetimepicker.js
+app/assets/javascripts/spectrum.js
+app/assets/javascripts/jquery.ba-throttle-debounce.min.js
+app/assets/javascripts/keycode.js.coffee
```
Comment (dabrau) [inline .eslintignore]: ignoring all the coffeescript in the linter and ignoring all the dependency modules, probably should move that to npm along with the js gems.

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    }
+  },
+  "env": {
+    "browser": true,
+    "node": true
+  },
+  "globals": {
```
Comment (dabrau) [inline .eslintrc.json]: tracking all the globals to be ignored by the linter. probably pay down at some point.

File: app/assets/javascripts/components/search.jsx [hunk 4]
```diff
@@ -307,7 +307,6 @@ var Search = React.createClass({
         )
         if (Object.keys(revisions).length > 0) {
           this.setState({mode: 'submit'})
-          console.log(revisions)
```
Comment (dabrau) [inline app/assets/javascripts/components/search.jsx]: removed all the logging. clutters the console.

File: app/assets/javascripts/components/timur.js.jsx [hunk 5]
```diff
@@ -1,6 +1,12 @@
+import createLogger from 'redux-logger'
```
Comment (dabrau) [inline app/assets/javascripts/components/timur.js.jsx]: added redux-logger for non-production builds.

Comment (graft) [inline app/assets/javascripts/components/timur.js.jsx]: How does NODE_ENV get set?

Comment (dabrau) [inline app/assets/javascripts/components/timur.js.jsx]: using the envify node package with browserify-rails should default the NODE_ENV to Rails.env automagically. https://github.com/browserify-rails/browserify-rails

**Comment by graft** [comment]
This is nice work. Important question: is there evidence this runs correctly? I will check it out tomorrow, if so it seems good to merge this.

**Comment by graft** [comment]
What's the alternative to all of these window.global_variable things? There must be a more ES6-y idiom to use here.

**Comment by dabrau** [comment]
Have them exportable in the files where they are defined and import where
they are used. I didn't think doing all those in one go would be worth it.
I figured we could just do that moving forward or if existing code gets
modified, that way its less likely to introduce bugs and hopefully keep
things in a relatively good state.

**Pull request merged**